src/net/encryption/src/key.rs

```diff
@@ -0,0 +1,42 @@
+use rsa::{RsaPublicKey, RsaPrivateKey, pkcs1::EncodeRsaPublicKey, Pkcs1v15Encrypt};
+use rand::rngs::OsRng;
+
+pub struct KeyPair {
+    pub public_key: RsaPublicKey,
+    pub encoded_public_key: der::Document,
+    pub private_key: RsaPrivateKey,
+}
```

Comment:
Some small documentation would be appreciated here

Reply:
I’ll fix the type and add some comments later tonight